python/pyspark/pandas/groupby.py

```diff
@@ -3534,7 +3534,12 @@ def _reduce_for_stat_function(
             for label in psdf._internal.column_labels:
                 psser = psdf._psser_for(label)
                 input_scol = psser._dtype_op.nan_to_null(psser).spark.column
-                output_scol = sfun(input_scol)
+                if sfun.__name__ == "sum" and isinstance(
+                    psdf._internal.spark_type_for(label), StringType
+                ):
+                    output_scol = F.concat_ws("", F.collect_list(input_scol))
```

Comment:
We should use combination of `concat_ws` and `collect_list` instead of `sum` to match the behavior with Pandas for string summation as below:

```python
>>> import pyspark.sql.functions as sf
>>> sdf.show()
+---+
|  A|
+---+
|  a|
|  b|
|  c|
+---+

# Using `sum` over string type column returns `NULL` which is not matched with pandas.
>>> sdf.select(sf.sum(sdf.A)).show()
+------+
|sum(A)|
+------+
|  NULL|
+------+

# Using combination of `concat_ws` and `collect_list` to match the pandas behavior
>>> sdf.select(sf.concat_ws("", sf.collect_list(sdf.A))).show()
+----------------------------+
|concat_ws(, collect_list(A))|
+----------------------------+
|                         abc|
+----------------------------+
```